### PR TITLE
[BAEL-3601] How to Lock Files in Java

### PR DESCRIPTION
**Pull request by psoares**
Fixes for PR 8587. For some reason my commits don't seem to amend PR 8587 automatically, so I'm submitting a new pull request.

### CODE REVIEW

File: core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import java.nio.channels.NonWritableChannelException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
```
Comment (JonCook) [inline core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java]: Let's remove this unused import

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.IOException;
+import java.nio.channels.FileLock;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+
+import org.junit.jupiter.api.Test;
+
+class FileLocksUnitTest {
+
+    @Test void givenAnInputStream_whenGetWriteLock_throwNonWritableChannelException() {
```
Comment (JonCook) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]: The formatting still looks to be off?

File: core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+public class FileLocks {
```
Comment (JonCook) [inline core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java]: Formatting also still looks to be off here in this class?

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java (hunk 2 above)
Comment (psoares) [reply core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]: Can you please tell me what is off? I used the intelliJ settings I found in https://github.com/eugenp/tutorials/blob/master/intelliJ/intelliJ-formatter.xml

File: core-java-modules/core-java-io-2/README.md [hunk 4]
```diff
@@ -11,5 +11,5 @@ This module contains articles about core Java input and output (IO)
 - [List Files in a Directory in Java](https://www.baeldung.com/java-list-directory-files)
 - [Java – Append Data to a File](https://www.baeldung.com/java-append-to-file)
 - [How to Copy a File with Java](https://www.baeldung.com/java-copy-file)
-- [Create a Directory in Java](https://www.baeldung.com/java-create-directory)
+- [Create a Directory in Java](https://www.baeldung.com/java-create-directory)
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-2/README.md]: no need to modify this file, please revert

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import java.nio.channels.FileLock;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+
+import org.junit.jupiter.api.Test;
+
+class FileLocksUnitTest {
+
+    @Test
+    void givenAnInputStream_whenGetWriteLock_throwNonWritableChannelException() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenAnInputStream_whenGetWriteLock_thenThrowNonWritableChannelException() {
```

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        assertThrows(NonWritableChannelException.class, () -> FileLocks.getExclusiveLockFromInputStream());
+    }
+
+    @Test
+    void givenARandomAccessFileWithReadWriteAccess_whenGetWriteLock_lock() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenARandomAccessFileWithReadWriteAccess_whenGetWriteLock_thenLock() throws IOException {
```

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        FileLock lock = FileLocks.getExclusiveLockFromRandomAccessFile(0, 10);
+        assertNotNull(lock);
+        assertFalse(lock.isShared());
+    }
+
+    @Test
+    void givenAPath_whenGetExclusiveLock_lock() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenAPath_whenGetExclusiveLock_thenLock() throws IOException {
```

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        FileLock lock = FileLocks.getExclusiveLockFromFileChannelOpen(0, 10);
+        assertNotNull(lock);
+        assertFalse(lock.isShared());
+    }
+
+    @Test
+    void givenAFileOutputStream_whenGetSharedLock_throwNonReadableChannelException() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenAFileOutputStream_whenGetSharedLock_thenThrowNonReadableChannelException() {
```

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        assertThrows(NonReadableChannelException.class, FileLocks::getReadLockFromOutputStream);
+    }
+
+    @Test
+    void givenAnInputStream_whenGetSharedLock_lock() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenAnInputStream_whenGetSharedLock_thenLock() throws IOException {
```

File: core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        FileLock lock = FileLocks.getReadLockFromInputStream(0, 10);
+        assertNotNull(lock);
+        assertTrue(lock.isShared());
+    }
+
+    @Test
+    void givenARandomAccessFile_whenGetSharedLock_lock() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio-2/src/test/java/com/baeldung/lock/FileLocksUnitTest.java]:
```suggestion
    void givenARandomAccessFile_whenGetSharedLock_thenLock() throws IOException {
```

**Comment by psoares** [comment]
Hi Kevin. I hope I resolved everything as expected. Let me know.

Thanks.

**Pull request merged**